src/app/static/src/app/components/projects/ProjectHistory.vue

```diff
@@ -114,90 +129,115 @@
-        data() {
-            return {
-                projectToDelete: null,
-                versionToDelete: null,
-                copiedProjectName: '',
-                projectToCopy: null,
-                versionToCopy: null,
-                shareProjectIsEnabled: switches.shareProject,
-                copyProjectIsEnabled: switches.copyProject
-            }
-        },
-        methods: {
-            format(date: string) {
-                return formatDateTime(date);
+    export default ProjectsMixin.extend<Data, Methods, Computed, {}>({
+       data() {
+           return {
+               projectToDelete: null,
+               versionToDelete: null,
+               versionToPromote: null,
+               newProjectName: "",
+               selectedVersionNumber: "",
+               shareProjectIsEnabled: switches.shareProject,
+               promoteProjectIsEnabled: switches.promoteProject
+           }
+       },
+        computed: {
+            disableCreate: function() {
+                return !this.newProjectName;
             },
-            loadVersion(event: Event, projectId: number, versionId: string) {
-                event.preventDefault();
-                this.loadAction({projectId, versionId});
+            promoteVersionHeader: function() {
+                return i18next.t('promoteVersionHeader', {version: this.selectedVersionNumber, lng: this.currentLanguage});
             },
-            deleteProject(event: Event, projectId: number) {
-                event.preventDefault();
-                this.projectToDelete = projectId;
-            },
-            deleteVersion(event: Event, projectId: number, versionId: string) {
-                event.preventDefault();
-                this.versionToDelete = {projectId, versionId};
-            },
-            copyProject(event: Event, projectId: number) {
-                event.preventDefault();
-                this.projectToCopy = projectId;
-            },
-            copyVersion(event: Event, projectId: number, versionId: string) {
-                event.preventDefault();
-                this.versionToCopy = {projectId, versionId};
-            },
-            cancelCopy() {
-                this.versionToCopy = null;
-                this.projectToCopy = null;
-            },
-            cancelDelete() {
-                this.versionToDelete = null;
-                this.projectToDelete = null;
-            },
-            confirmDelete() {
-                if (this.projectToDelete) {
-                    this.deleteProjectAction(this.projectToDelete);
-                    this.projectToDelete = null;
-                } else if (this.versionToDelete) {
-                    this.deleteVersionAction(this.versionToDelete);
-                    this.versionToDelete = null;
+            currentLanguage: mapStateProp<RootState, Language>(null,
+                (state: RootState) => state.language)
+        },
```

Comment:
Something a bit funny going on with spacing, can you auto-format this file? (sorry for nit-picking, we should hopefully soon have a linter for the front-end that will automatically pick these things up on Travis!)

Reply:
Oh dear, the spacing here is still a bit off, but don't worry about it. We need to commit a style rules file and linter that enforces styles properly.